lib/Models/SearchResult.ts

```diff
@@ -23,6 +24,7 @@ export default class SearchResult {
   @observable isOpen = false;
   @observable type: string = "search_result";
   @observable location: { longitude: number; latitude: number } | undefined;
+  @observable id: number | undefined;
```

Comment:
I think to be more general the id should be `string | undefined`. Across the rest of Terria ids are predominately `string`s. And any `number` can be cast to a `string`, but not the other way around.

```suggestion
  @observable id: string | undefined;
```